contracts/DeresyResolver.sol

```diff
@@ -267,32 +275,34 @@ contract DeresyResolver is SchemaResolver, Ownable {
   function createReviewRequestCommon(
       string memory _name,
       address[] memory reviewers,
+      address[] memory reviewerContracts,
       uint256[] memory hypercertIDs,
       string[] memory hypercertIPFSHashes,
       string memory formIpfsHash,
       uint256 rewardPerReview,
+      uint8 reviewsPerHypercert,
       string memory reviewFormName,
       address paymentTokenAddress,
       bool isPayable
   ) internal {
     require(bytes(_name).length > 0, "Deresy: RequestName cannot be null or empty");
-    require(reviewers.length > 0, "Deresy: Reviewers cannot be null");
+    require(reviewers.length > 0 || reviewerContracts.length > 0, "Deresy: Reviewers cannot be null");
     require(hypercertIDs.length > 0, "Deresy: Hypercert IDs cannot be null");
     require(hypercertIPFSHashes.length > 0, "Deresy: Hypercerts IPFS hashes cannot be null");
     require(hypercertIDs.length == hypercertIPFSHashes.length, "Deresy: HypercertIDs and HypercertIPFSHashes array must have the same length");
     require(reviewForms[reviewFormName].questions.length > 0, "Deresy: ReviewFormName does not exist");
     require(reviewRequests[_name].sponsor == address(0),"Deresy: Name duplicated");
     require(isTokenWhitelisted(paymentTokenAddress),"Deresy: Token is not whitelisted");
     require(validateRequestHypercerts(hypercertIDs), "Deresy: One or more hypercerts are not valid");
+    require(allReviewerContractsERC721(reviewerContracts), "Deresy: Not all reviewer contracts are ERC721 contracts");
 
     uint256 tokenFundsAmount;
 
     if (isPayable) {
       require(rewardPerReview > 0, "Deresy: rewardPerReview must be greater than zero for payable request");
 
       if (paymentTokenAddress == address(0)) {
-        require(msg.value >= ((reviewers.length * hypercertIDs.length) * rewardPerReview), "Deresy: msg.value invalid");
-
+        require(msg.value >= ((reviewsPerHypercert * hypercertIDs.length) * rewardPerReview), "Deresy: msg.value invalid");
         tokenFundsAmount = msg.value;
       } else {
         require(msg.value == 0, "Deresy: msg.value is invalid");
```

Comment:
The else condition needs the same change as above (ie use reviewsPerHypercert instead of reviewers.length. Please fix & test.

Reply:
Let's add a non-regression test that covers this case as well.